### PR TITLE
Switch `truffle test` to use new compile interface

### PR DESCRIPTION
**Pull request by gnidan**
Implements #2211 
Builds on #2073 #2258 

- Use `"truffle-workflow-compile/new"` inside lib/test
- ~~Remove extraneous profiler usage (truffle-workflow-compile does this)~~ turns out this isn't so extraneous
- Capture entire compilation result instead of just dependencyPaths

### CODE REVIEW

File: packages/truffle-core/lib/test.js [hunk 1]
```diff
@@ -144,36 +145,33 @@ const Test = {
...
-        // Compile project contracts and test contracts
-        Contracts.compile(compileConfig)
-          .then(result => {
-            const paths = result.outputs.solc;
-            accept(paths);
-          })
-          .catch(reject);
-      });
+    const updated =
+      (await promisify(Profiler.updated)(
```
Comment (eggplantzzz) [inline packages/truffle-core/lib/test.js]: No need for this `promisify`. `Profiler.updated` now has a dual Promise/callback interface. Check it [out](https://github.com/trufflesuite/truffle/blob/develop/packages/truffle-compile/profiler/index.js#L19).

File: packages/truffle-core/lib/test.js [hunk 2]
```diff
@@ -144,36 +145,33 @@ const Test = {
...
+        config.with({ resolver: testResolver })
+      )) || [];
+
+    const compileConfig = config.with({
+      all: config.compileAll === true,
+      files: updated.concat(solidityTestFiles),
+      resolver: testResolver,
+      quiet: false,
```
Comment (eggplantzzz) [inline packages/truffle-core/lib/test.js]: Don't we want this to come from the config? Wasn't this something changed recently by @CruzMolina?

File: packages/truffle-core/lib/test.js [hunk 3]
```diff
@@ -101,7 +102,7 @@ const Test = {
     await this.defineSolidityTests(
       mocha,
       testContracts,
-      dependencyPaths,
+      compilations.solc.sourceIndexes,
```
Comment (eggplantzzz) [inline packages/truffle-core/lib/test.js]: Is this the same thing as `dependencyPaths` was??

File: packages/truffle-core/lib/test.js (hunk 1 above)
Comment (gnidan) [reply packages/truffle-core/lib/test.js]: Hot diggity!

File: packages/truffle-core/lib/test.js (hunk 2 above)
Comment (gnidan) [reply packages/truffle-core/lib/test.js]: Oh yeah you're right!

File: packages/truffle-core/lib/test.js (hunk 3 above)
Comment (gnidan) [reply packages/truffle-core/lib/test.js]: Yep

Comment (eggplantzzz) [reply packages/truffle-core/lib/test.js]: Rad

**Comment by gnidan** [comment]
Thanks!

**Pull request merged**